airbyte-integrations/connectors/source-mongodb/lib/mongodb_types_explorer.rb

```diff
@@ -5,30 +5,36 @@
 class MongodbTypesExplorer
   EXPLORE_LIMIT = 1_000
 
-  attr_reader :field_type
+  @@cache = {}
 
-  def initialize(collection:, field:, limit: EXPLORE_LIMIT, &type_mapping_block)
-    @collection = collection
-    @field = field
-    @limit = limit
-    @type_mapping_block = type_mapping_block
+  def self.run(collection:, field:, limit: EXPLORE_LIMIT, &type_mapping_block)
+    determine_field_types_for_collection(collection: collection, limit: limit, &type_mapping_block)
 
-    @field_type = determine_field_type
+    @@cache[collection.name][field]
   end
 
   private
 
-  def determine_field_type
-    airbyte_types = Set[]
+  def self.determine_field_types_for_collection(collection:, limit:, &type_mapping_block)
+    return if @@cache[collection.name]
 
-    @collection.find(@field => { "$nin": [nil] }).limit(@limit).each do |item|
-      mapped_value = @type_mapping_block[item[@field].class]
-      airbyte_types.add(mapped_value)
+    airbyte_types = {}
+
+    collection.find.limit(limit).each do |item|
+      item.each_pair do |key, value|
+        mapped_value = type_mapping_block[value.class]
+
+        airbyte_types[key] ||= Set[]
+        airbyte_types[key].add(mapped_value)
+      end
     end
 
-    # Has one specific type
-    if airbyte_types.count == 1
-      airbyte_types.first
+    @@cache[collection.name] = {}
+    airbyte_types.each_pair do |field, types|
+      # Has one specific type
+      if types.count == 1
```

Comment:
does it make sense to have a fallback type here in case there are more than one type? we typically use `string` as the type in other connectors